src/cargo/sources/registry/index.rs

```diff
@@ -446,15 +489,21 @@ impl<'cfg> RegistryIndex<'cfg> {
             f(summary);
             count += 1;
         }
-        Ok(count)
+        Ok(Poll::Ready(count))
     }
 
     pub fn is_yanked(&mut self, pkg: PackageId, load: &mut dyn RegistryData) -> CargoResult<bool> {
         let req = VersionReq::exact(pkg.version());
-        let found = self
-            .summaries(pkg.name(), &req, load)?
-            .any(|summary| summary.yanked);
-        Ok(found)
+        loop {
```

Comment:
Should this just also be modified to return `Poll<bool>`?

Reply:
One level is straightforward and worth it. Continuing up the stack `Source::is_yanked`; I am not seeing where we would want to call a lot of it in parallel nor where it ends in an existing loop. Seams like it is called where there is not an opportunity for parallelism (`install` code) or where we have already updated that pkg (writing a `lockfile`). What have I missed?